src/sage/interfaces/ecm.py

```diff
@@ -55,6 +55,7 @@
 from sage.structure.sage_object import SageObject
 from sage.rings.integer_ring import ZZ
 
+from sage.env import ECM
```

Comment:
Should you get the filename from the feature instead? This way it seems it will only work if `ECM` is set, there's no fallback if `ECM` is None.

Reply:
Yes, it should be using the Feature.

Reply:
I have no idea how to use `Feature` here. It seems to provide absolute executable names, but no basenames.
Perhaps its a missing feature in `Feature`.

I am certainly not going to implement `basename` in Python just here.

Reply:
Why do you think you need the basename?

Reply:
Because this is what `sage.interfaces.ecm` is using. 
Not sure why whoever wrote this got so creative, but that's how it is.

Reply:
What are you talking about. It's just using `Popen`, of course you can pass the full executable name to it, not just the basename.

Reply:
> I have no idea how to use `Feature` here. It seems to provide absolute executable names, but no basenames. Perhaps its a missing feature in `Feature`.

Using the full path seems better, to make sure there's a single point where the decision of which executable to use is made. How the feature found the path, whether it's static or dynamic, if we want to support PATH being different at build time than at run time, etc, is all abstracted out in the feature.

> I am certainly not going to implement `basename` in Python just here.

Just for the record, there's `os.path.basename()` (and `os.path.dirname()`, and `os.path.split()`)